File: lemur-libs/lemur-contract/src/main/java/io/github/malczuuu/lemur/contract/rest/PageResult.java
package io.github.malczuuu.lemur.contract.rest;

import io.github.malczuuu.lemur.contract.TransportResponse;
import java.util.List;

public record PageResult<T>(
    List<T> content, Long page, Long size, Long totalElements, Long totalPages)
    implements TransportResponse {}

File: lemur-libs/lemur-contract/src/main/kotlin/io/github/malczuuu/lemur/contract/rest/PageResult.kt
package io.github.malczuuu.lemur.contract.rest

import io.github.malczuuu.lemur.contract.TransportResponse
import jakarta.validation.constraints.PositiveOrZero

data class PageResult<T>(
    val content: List<T>,

    @field:PositiveOrZero
    val page: Long,

    @field:PositiveOrZero
    val size: Long,

    @field:PositiveOrZero
    val totalElements: Long,

    @field:PositiveOrZero
    val totalPages: Long,
) : TransportResponse
